### PR TITLE
Integrate llvm-project at 6c6dff7e2c27 and bump dependencies

### PR DESCRIPTION
**Pull request by hanhanW**
* Reset third_party/llvm-project: 6c6dff7e2c27c5a9ea9466d49f61a1edc82bc364 (2022-11-04 23:20:37 -0400): [libc] Add add_with_carry to builtin wrapper.
* mlir-hlo: 4cb08f5b1d612c2d9bc6011aaf73bd0141b7e5fa
* tensorflow: cdb73b19739ccd72fcb659d220bf71924c056ced

Cherry-pick https://github.com/llvm/llvm-project/commit/7c2d3153a9481793da58894dbf35d4994f3b67a4

### CODE REVIEW

**Comment by hanhanW** [comment]
@antiagainst @kuhar There are a lot of failures in the PR. One of them are about vulkan-spirv, I found that we're not able to compile softmax though vulkan-spirv backend and I already narrow it down to `spirv-canonicalize-gl`. Can one of you help fix `spirv-canonicalize-gl` issue?

To repro:

`bin/mlir-opt -spirv-canonicalize-gl ~/spirv.mlir`.

https://gist.githubusercontent.com/hanhanW/32db5c54549cb12793ffac430bef7db7/raw

In the meantime, I'll be fixing other failures.

**Comment by antiagainst** [comment]
Thanks @hanhanW for isolating the issue! Let me take a look..

**Comment by antiagainst** [comment]
https://github.com/llvm/llvm-project/commit/7c2d3153a9481793da58894dbf35d4994f3b67a4 should address the issue.

**Review by antiagainst — APPROVED** [review]
